### PR TITLE
fix(autofix): Fix issues with feedback button in autofix drawer

### PR DESCRIPTION
**Pull request by malwilley**
Fixes a couple issues:

- The feedback dialog would close whenever useAutofix polled for new data
- Interacting with the feedback dialog would close the drawer

### CODE REVIEW

File: static/app/components/events/autofix/autofixBanner.tsx [hunk 1]
```diff
@@ -41,7 +41,10 @@ function SuccessfulSetup({
         // we don't reopen it immediately, and instead let the button handle this itself.
         shouldCloseOnInteractOutside: element => {
           const viewAllButton = openButtonRef.current;
-          if (viewAllButton?.contains(element)) {
+          if (
+            viewAllButton?.contains(element) ||
+            document.getElementById('sentry-feedback')?.contains(element)
```
Comment (malwilley) [inline static/app/components/events/autofix/autofixBanner.tsx:46]: This prevents the drawer from closing when interacting with the feedback dialog

File: static/app/components/events/autofix/autofixFeedback.tsx [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
-    buttonRef,
-    messagePlaceholder: t('How can we make Autofix better for you?'),
-    optionOverrides: {
-      tags: {
-        ['feedback.source']: 'issue_details_ai_autofix',
-        ['feedback.owner']: 'ml-ai',
-      },
-    },
-  });
+  const openForm = useFeedbackForm();
```
Comment (malwilley) [inline static/app/components/events/autofix/autofixFeedback.tsx:10]: This uses the new `useFeedbackForm` hook which uses a form rendered at the root so it doesn't have the same issue as the other way

File: static/app/views/app/index.tsx [hunk 3]
```diff
@@ -246,16 +246,16 @@ function App({children, params}: Props) {
         <RouteAnalyticsContextProvider>
           <OrganizationContextProvider>
             <AsyncSDKIntegrationContextProvider>
-              <GlobalDrawer>
-                <GlobalFeedbackForm>
+              <GlobalFeedbackForm>
```
Comment (malwilley) [inline static/app/views/app/index.tsx:249]: Had to lift the feedback from context one level so that the drawer component could access it

**Review by roaga — APPROVED** [review]
Looks good to me, though maybe I'm not the best person to review the non-Autofix change. Thanks for fixing this!

**Pull request merged**